beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -1893,6 +1895,111 @@ func (s *Server) GetProposerLookahead(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// GetBlobs retrieves blobs for a given block id.
+func (s *Server) GetBlobs(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetBlobs")
+	defer span.End()
+
+	segments := strings.Split(r.URL.Path, "/")
+	blockId := segments[len(segments)-1]
+
+	var verifiedBlobs []*blocks.VerifiedROBlob
+	var rpcErr *core.RpcError
+
+	// Check if versioned_hashes parameter is provided
+	versionedHashesStr := r.URL.Query()["versioned_hashes"]
+	versionedHashes := make([][]byte, 0, len(versionedHashesStr))
+	if len(versionedHashesStr) > 0 {
+		for _, hashStr := range versionedHashesStr {
+			hash, err := hexutil.Decode(hashStr)
+			if err != nil {
+				httputil.HandleError(w, fmt.Sprintf("Invalid versioned hash %s: %v", hashStr, err), http.StatusBadRequest)
+				return
+			}
+			if len(hash) != 32 {
+				httputil.HandleError(w, fmt.Sprintf("Invalid versioned hash length for %s: expected 32 bytes, got %d", hashStr, len(hash)), http.StatusBadRequest)
+				return
+			}
+			versionedHashes = append(versionedHashes, hash)
+		}
+	}
+	verifiedBlobs, rpcErr = s.Blocker.Blobs(ctx, blockId, options.WithVersionedHashes(versionedHashes))
+	if rpcErr != nil {
+		code := core.ErrorReasonToHTTP(rpcErr.Reason)
+		switch code {
+		case http.StatusBadRequest:
+			httputil.HandleError(w, "Invalid block ID: "+rpcErr.Err.Error(), code)
+			return
+		case http.StatusNotFound:
+			httputil.HandleError(w, "Block not found: "+rpcErr.Err.Error(), code)
+			return
```

Comment:
```suggestion
		case http.StatusBadRequest:
			httputil.HandleError(w, "Bad request: "+rpcErr.Err.Error(), code)
			return
		case http.StatusNotFound:
			httputil.HandleError(w, "Not found: "+rpcErr.Err.Error(), code)
			return
```
Because there can be something wrong with either the blocks or blobs